src/Host-SDK/Temporal.Worker.Hosting/HostingApi.cs

```diff
@@ -0,0 +1,223 @@
+        {
+            return hostBuilder;
+        }
+
+        public static IHostBuilder UseTemporalWorkerHost(this IHostBuilder hostBuilder,
+                                                         Action<HostBuilderContext, TemporalServiceConfiguration> configureTemporalService)
+        {
+            return hostBuilder;
+        }
+
+        public static IHostBuilder UseTemporalWorkerHost(this IHostBuilder hostBuilder,
+                                                         Action<TemporalServiceConfiguration,
+                                                                TemporalWorkerConfiguration,
+                                                                WorkflowExecutionConfiguration,
+                                                                WorkflowImplementationConfiguration> configureDefaults)
+        {
+            return hostBuilder;
+        }
+
+        public static IHostBuilder UseTemporalWorkerHost(this IHostBuilder hostBuilder,
+                                                        Action<HostBuilderContext,
+                                                               TemporalServiceConfiguration,
+                                                               TemporalWorkerConfiguration,
+                                                               WorkflowExecutionConfiguration,
+                                                               WorkflowImplementationConfiguration> configureDefaults)
+        {
+            return hostBuilder;
+        }
+    }
+
+    public interface ITemporalServiceConfiguration
+    {
+        string OrchestratorServiceUrl { get; }
+        string Namespace { get; }
+    }
+
+    public class TemporalServiceConfiguration : ITemporalServiceConfiguration
+    {
+        public string OrchestratorServiceUrl { get; set; }
+        public string Namespace { get; set; }
+    }
+
+    // ----------- -----------
+
+    public static class ServiceCollectionExtensions
+    {
+        public static WorkerRegistration AddTemporalWorker(this IServiceCollection serviceCollection)
+        { return null; }        
+
+        public static WorkflowRegistration AddWorkflowWithOverrides<TWorkflowImplementation>(this IServiceCollection serviceCollection)
+                where TWorkflowImplementation : class, IBasicWorkflow
+        { return null; }
+
+        public static WorkflowRegistration AddWorkflowWithAttributes<TWorkflowImplementation>(this IServiceCollection serviceCollection)
+                where TWorkflowImplementation : class
+        { return null; }
+
+        public static ActivityRegistration AddActivity<TActivityImplementation>(this IServiceCollection serviceCollection)
+                where TActivityImplementation : class, IBasicActivity
+        { return null; }
+
+        public static ActivityRegistration AddActivity(this IServiceCollection serviceCollection,
+                                                       string activityTypeName,
```

Comment:
Why does registration take a string here when the activity class define the activity type name already?

Reply:
This overload allows specifying any delegate. The overload that takes an IBasicActivity does not require the name

Reply:
I don't find this especially convenient and I'd avoid using strings at all cost, they're too error prone.